lib/node_modules/@stdlib/lapack/base/dlarfg/lib/base.js

```diff
@@ -0,0 +1,116 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2025 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var dnrm2 = require( '@stdlib/blas/base/dnrm2' ).ndarray;
+var sign = require( '@stdlib/math/base/special/copysign' );
+var dlamch = require( '@stdlib/lapack/base/dlamch' );
+var abs = require( '@stdlib/math/base/special/abs' );
+var dscal = require( '@stdlib/blas/base/dscal' ).ndarray;
+var dlapy2 = require( '@stdlib/lapack/base/dlapy2' );
+
+
+// MAIN //
+
+/**
+* Generates a real elementary reflector `H` of order `N` such that applying `H` to a vector `[alpha; x]` zeros out `X`.
+*
+* ## Notes
+*
+* -   `H` is a Householder matrix with the form `H = I - tau * [1; v] * [1, v^T]`, where `tau` is a scalar and `v` is a vector.
+* -   the input vector is `[alpha; x]`, where `alpha` is a scalar and `X` is a real `(n-1)`-element vector.
+* -   the result of applying `H` to `[alpha; x]` is `[beta; 0]`, with `beta` being a scalar and the rest of the vector zeroed.
+* -   if all elements of `X` are zero, then `tau = 0` and `H` is the identity matrix.
+* -   otherwise, `1 <= tau <= 2`
+*
+* @private
+* @param {NonNegativeInteger} N - number of rows/columns of the elementary reflector `H`
+* @param {Float64Array} X - overwritten by the vector `V` on exit, expects `N - 1` indexed elements
+* @param {integer} strideX - stride length for `X`
+* @param {NonNegativeInteger} offsetX - starting index of `X`
+* @param {Float64Array} out - array to store `alpha` and `tau`, first indexed element stores `alpha` and the second indexed element stores `tau`
```

Comment:
```suggestion
* -   otherwise, `1 <= tau <= 2`
* -   `X` should have `N-1` indexed elements.
* -   The output array contains the following two elements: `alpha` and `tau`.
*
* @private
* @param {NonNegativeInteger} N - number of rows/columns of the elementary reflector `H`
* @param {Float64Array} X - input vector
* @param {integer} strideX - stride length for `X`
* @param {NonNegativeInteger} offsetX - starting index of `X`
* @param {Float64Array} out - output array
```